app/controllers/general.php

```diff
@@ -300,6 +301,35 @@
 
                 Authorization::setRole('role:' . Auth::USER_ROLE_APP);
                 Authorization::setDefaultStatus(false);  // Cancel security segmentation for API keys.
+
+                if (time() > ($key->getAttribute('accessedAt', 0) + (24 * 60 * 60))) {
+                    $key->setAttribute('accessedAt', time());
+                    $sdkValidator = new WhiteList([
+                        'appwrite:nodejs',
+                        'appwrite:deno',
+                        'appwrite:php',
+                        'appwrite:python',
+                        'appwrite:ruby',
+                        'appwrite:go',
+                        'appwrite:java',
+                        'appwrite:dotnet',
+                        'appwrite:dart',
+                        'appwrite:kotlin',
+                        'appwrite:swift'
+                    ], true);
+    
+                    $sdk = $request->getHeader('x-sdk-version', 'UNKNOWN');
+                    $sdk = substr($sdk, 0, strrpos($sdk, ':'));
+
+                    if ($sdkValidator->isValid($sdk)) {
+                        $sdks = $key->getAttribute('sdks', []);
+                        array_push($sdks, $sdk);
```

Comment:
This method will get the array bigger every time.

Do we also want to separate the 24 hours check and this check? Otherwise, we could potentially miss some SDKs. We could perform writes once every 24 hours for the access time, and every time a new (allowed) SDK is discovered. This could ensure min writes and max data integrity.

Reply:
> This method will get the array bigger every time.
good catch. Fixed it 👍

Reply:
> We could perform writes once every 24 hours for the access time, and every time a new (allowed) SDK is discovered. This could ensure min writes and max data integrity.

If we want to write every time a new SDK is used, it makes sense to update the access time as well. Because its a single write operation after all

Reply:
We can do that.